win32/OpenSC.wxs.in

```diff
@@ -1,462 +1,255 @@
-                  <RegistryValue Type="string" Name="Image Path" Value="basecsp.dll" KeyPath="yes"/>
-                  <RegistryValue Type="integer" Name="Type" Value="1"/>
-                </RegistryKey>
-                <!-- when the x64 installer will install x86 components on x64 plateform too
-                    <?if $(sys.BUILDARCH) = x64 ?>
-                    <RegistryKey Root="HKLM" Key="SOFTWARE\Wow6432Node\Microsoft\Cryptography\Defaults\Provider\OpenSC CSP">
-                      <RegistryValue Type="string" Name="Image Path" Value="basecsp.dll"/>
-                      <RegistryValue Type="integer" Name="Type" Value="1"/>
-                    </RegistryKey>
-                    <?endif?>
-                    -->
-              </Component>
-              <Component Id="CertPropSvc" Guid="*" Win64="$(var.Win64YesNo)" Permanent="yes">
-                <!-- CertPropSvc loads the minidriver and propagates the smart card's
-                     certificates to the user's certificate store, see https://technet.microsoft.com/en-us/itpro/windows/keep-secure/smart-card-certificate-propagation-service -->
-                <ServiceControl Id="ControlCertPropSvcStart" Name="CertPropSvc" Start="install" Wait="no" />
-                <ServiceControl Id="ControlCertPropSvcStop" Name="CertPropSvc" Stop="uninstall" Wait="yes" />
-                <RegistryKey Root="HKLM" Key="SYSTEM\CurrentControlSet\Services\CertPropSvc">
-                  <RegistryValue Type="integer" Name="Start" Value="2" KeyPath="yes"/>
-                </RegistryKey>
-              </Component>
-            </Directory>
+          <Directory Id="INSTALLDIR_PKCS11" Name="pkcs11">
+            <Component Id="opensc_pkcs11.dll">
+              <File Source="$(var.SOURCE_DIR)\src\pkcs11\opensc-pkcs11.dll" />
+            </Component>
+            <Component Id="onepin_opensc_pkcs11.dll">
+              <File Name="onepin-opensc-pkcs11.dll" Source="$(var.SOURCE_DIR)\src\pkcs11\opensc-pkcs11.dll" />
+            </Component>
+          </Directory>
 
-            <Directory Id="INSTALLDIR_PKCS11" Name="pkcs11">
-              <Component Id="opensc_pkcs11.dll" Guid="*" Win64="$(var.Win64YesNo)">
-                <File Source="$(var.SOURCE_DIR)\src\pkcs11\opensc-pkcs11.dll" Vital="yes"/>
-              </Component>
-              <Component Id="onepin_opensc_pkcs11.dll" Guid="*" Win64="$(var.Win64YesNo)">
-                <File Name="onepin-opensc-pkcs11.dll" Source="$(var.SOURCE_DIR)\src\pkcs11\opensc-pkcs11.dll" Vital="yes"/>
-              </Component>
-            </Directory>
+          <Component Id="Autostart_tools">
+            <RegistryKey Root="HKMU" Key="Software\Microsoft\Windows\CurrentVersion\Run">
+              <RegistryValue Type="string" Name="pkcs11-register.exe" Value="[INSTALLDIR_TOOLS]pkcs11-register.exe" />
+            </RegistryKey>
+          </Component>
+          <Component Id="Autostart_native_tools" Condition="NOT Msix64">
```

Comment:
@frankmorgner Do you have any hint what this condition does?

Reply:
Notifications are initialized on login as well as pkcs11-register. The idea in the original script was that it would only start if run in a native envirnment. For example, the 32 bit notification service should not initialize in a 64 bit system to avoid running two executables.

Reply:
The converter messed up then this condition, I will try to fix it

Reply:
Fixed